src/apexguru/apex-guru-service.ts

```
@@ -29,22 +33,73 @@ export async function isApexGuruEnabledInOrg(outputChannel: vscode.LogOutputChan
+			outputChannel.appendLine('***Apex Guru request Id:***' + requestId);
+
+			const queryResponse: ApexGuruQueryResponse = await pollAndGetApexGuruResponse(connection, requestId, Constants.APEX_GURU_MAX_TIMEOUT_SECONDS, Constants.APEX_GURU_RETRY_INTERVAL_MILLIS);
+
+			const decodedReport = Buffer.from(queryResponse.report, 'base64').toString('utf8');
+			outputChannel.appendLine('***Retrieved analysis report from ApexGuru***:' + decodedReport);
+
+			const ruleResult = transformStringToRuleResult(selection.fsPath, decodedReport);
+			new DiagnosticManager().displayDiagnostics([selection.fsPath], [ruleResult], diagnosticCollection);
+			TelemetryService.sendCommandEvent(Constants.TELEM_SUCCESSFUL_APEX_GURU_FILE_ANALYSIS, {
+				executedCommand: commandName,
+				duration: (Date.now() - startTime).toString()
+			});
+		});
     } catch (e) {
         const errMsg = e instanceof Error ? e.message : e as string;
         outputChannel.appendLine('***Apex Guru initiate request failed***');
         outputChannel.appendLine(errMsg);
     }
 }
 
-export async function initiateApexGuruRequest(selection: vscode.Uri, outputChannel: vscode.LogOutputChannel): Promise<string> {
+export async function pollAndGetApexGuruResponse(connection: Connection, requestId: string, maxWaitTimeInSeconds: number, retryIntervalInMillis: number) {
+	let queryResponse: ApexGuruQueryResponse;
+	const startTime = Date.now();
+	while ((Date.now() - startTime) < maxWaitTimeInSeconds * 1000) {
+		try {
+			queryResponse = await connection.request({
+				method: 'GET',
+				url: `${Constants.APEX_GURU_REQUEST}/${requestId}`,
+				body: ''
+			});
+
+			if (queryResponse.status == 'success') {
+				return queryResponse;
+			} else {
+				// Add a delay between requests
+				await new Promise(resolve => setTimeout(resolve, retryIntervalInMillis));
+			}
+		} catch (error) {
+            await new Promise(resolve => setTimeout(resolve, retryIntervalInMillis));
+        }
```

I wouldn't want to lose this error. I don't think we need a catch statement here do we? Do we care to do retries on error? I didn't think we cared. But if we do, then let's not lose this error. Store it as the last error found... so that if we reach line 95 we can throw the actual error.